### DISCUSSION TITLE
Import/Export success rate is very unstable

### DISCUSSION THREAD

**Discussion by toreriksolem** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#69] [Editor App Feature Requests]
80+ % of the times I want to "send to device", the spinning circle with "sending..." inside it just keeps on spinning forever. 

Maybe it would be a better solution to use a responsive solution where we can see how the syncing is progressing. This would probably make me more impatient compared to how it currently works, as it actually is not giving any actual feedback on how the progress is doing. 

MacBook
 OS 11.5.1 
Running web app in Chrome Version 104.0.5112.79
USB-C cable directly to Bridge 6

**Comment by simonaglover95** [comment]
Mine "mostly" works, but I also occasionally get problems with either "get from device" or "send to device" 

Nothing is different in the process, it just sometimes continues "send" or "importing" with the spinning wheel on the app, but never finishes. 

Attached is the config and the console output from the last session when this happened. It froze on "sending to device" and never completed.
<img width="1027" alt="Screen Shot 2022-08-19 at 4 42 24 pm" src="https://user-images.githubusercontent.com/59382082/185611471-01b35076-005e-46f2-8a57-ba8cd1f544a1.png">
<img width="718" alt="Screen Shot 2022-08-19 at 4 42 53 pm" src="https://user-images.githubusercontent.com/59382082/185611495-a784ec07-bb6f-4a3a-9b74-8202581492e9.png">
<img width="736" alt="Screen Shot 2022-08-19 at 4 43 38 pm" src="https://user-images.githubusercontent.com/59382082/185611498-05caf3c6-cdef-4f2d-a7f8-4a8aafcaf5f9.png">

**Comment by konnsim** [comment]
I was having a hard time replicating this, but I've uploaded what is potentially a fix in the form of short delays during device communication.
Can you please give it a test and let me know how you go.

**Comment by konnsim** [comment]
@toreriksolem @simonaglover95 Just touching base to see if you have had any more of the same issue since I posted. If not I'll close this off.

**Reply by toreriksolem** [reply]
I am Sorry, I haven't had the time these two weeks to follow up. Tomorrow might be the day for it.

**Reply by simonaglover95** [reply]
I have done some limited testing with no issues :) haven't gone crazy, but I suspect the issue is fixed.

**Reply by konnsim** [reply]
@toreriksolem All good no rush just keen for your feedback. Let me know how you go when you get a chance.
@simonaglover95 Good to hear.

**Reply by simonaglover95** [reply]
@Konnsalad I was able to get the BRIDGE4 to break this morning. Changed bank 1 name, sent to device no problem, then change the global Flexiport modes to 1=Dual Exp and 2=Switch In, and it didn't succeed. Hung on the "sending" screen. :'(

If I make it happen again, I'll save the config for you.

**Reply by simonaglover95** [reply]
but i haven't made it happen on BRIDGE6 yet and I've spent more time trying to break that one.

**Reply by simonaglover95** [reply]
So, After the failed transfer, the without doing anything to the BRIDGE4, it seems to be unable to respond to the CHCK command. See screenshot attached. Must have been frozen up from the last hang. 
<img width="707" alt="Screen Shot 2022-09-06 at 9 14 55 am" src="https://user-images.githubusercontent.com/59382082/188545777-074ed14a-dbd0-4ae4-a1d8-e67915c0108e.png">

**Reply by simonaglover95** [reply]
doesn't seem to be directly related to what setting I change, but it has consistently done it on the second consecutive "send to device" 
Basically I plug it in fresh, change a couple of things, send to device - it works fine. Then I don't do anything except change a few more settings and send to device again - then it hangs. 
<img width="730" alt="Screen Shot 2022-09-06 at 9 20 58 am" src="https://user-images.githubusercontent.com/59382082/188546431-a9f147ff-2007-4422-acec-d2a4735739cb.png">
